File: Java/Strings/checkStringUnique.java
// Determine if a string has unique characters without using additional data structures

import java.util.Arrays;

public class Main {
  boolean uniqueCharacters(String str){
    boolean result = false;
    char[] strArr = str.toCharArray();
    Arrays.sort(strArr); 
    for(char c: strArr){
      if(str.indexOf(c) != str.lastIndexOf(c)){
          result = false;
      }else{
        result = true;
      }
    }
    return result;
  }
}

File: Javascript/Strings/checkStringUnique.js
// Determine if a string has unique characters without using additional data structures

function checkStringUnique(str) {
  if (str === "") {
    return false;
  }
  for (let i = 0; i < str.length; i++) {
    if (str.indexOf(str[i]) !== str.lastIndexOf(str[i])) {
      return false;
    }
  }
  return true;
}

// checkStringUnique('');
// checkStringUnique('apple');
// checkStringUnique('unique');
